src/store/local-storage-events-emitter.ts

```diff
@@ -0,0 +1,29 @@
+/*
+Copyright 2021 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { TypedEventEmitter } from "../models/typed-event-emitter";
+
+export enum LocalStorageErrors {
+    Global = 'Global',
+    SetItemError = 'setItem',
+    GetItemError = 'getItem',
+    RemoveItemError = 'removeItem',
+    ClearError = 'clear',
+    QuotaExceededError = 'QuotaExceededError'
+}
+
+// Used in element-web as a temporary hack to handle all the localStorage errors on the highest level possible
```

Comment:
It would be interesting to capture the goal we're aiming for and reference the issue here.
If I was to come across this comment, I would be unsure about why this is hacky and what we're trying to achieve

Reply:
@gsouquet that's why we have git blame! 😄 
I'll add the issue here though, good idea.

Reply:
See the comment now, should be much better.

Reply:
The comment does not address why this is a temporary hack nor the ideal solution we're aiming for

The git blame did not mention any of that before you force pushed...